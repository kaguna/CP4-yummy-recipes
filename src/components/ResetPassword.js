import React, { Component } from 'react';
import axiosInstance from '../AxiosInstance'
import Home from './Home';
class ResetPassword extends Component {
    constructor(props){
        super(props);
        this.state = { email: ""}
    }

    inputHandler = (event) => {
        const {name, value} = event.target;
        this.setState({[name]: value});
    };

    resetPasswordHandler = (event) => {
        const {email} = this.state;
        event.preventDefault();
        axiosInstance.post("/auth/send_reset_password_token", {email})
            .then(response => {
                this.setState({mess: response.data.message, error: "",});
            })
            .catch(error => {
                if (error.response) {
                    this.setState({error: error.response.data.message, mess: ""});
                } else if (error.request) {
                    this.setState({error:"Can't connect to the server.Please check your connection and try again.", mess: ""});
                }
            });
    };
    render() {
        return (
            <div className="modal fade" id="resetpassword" tabindex = "-1" role="dialog"
            aria-labelledby="exampleModalCenterTitle" aria-hidden="true">
           <div className="modal-dialog" role="document">
               <div className="modal-content">
                   <div class="modal-header">
                       <h5 className="modal-title" id="exampleModalLongTitle">Email login credetials</h5>
                       <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                           <span aria-hidden="true">&times;</span>
                       </button>
                   </div>
                   <div className="modal-body">
                   {this.state.mess?
                                <div className="alert alert-success">{this.state.mess}</div>: ""}
                                {this.state.error?
                                <div className="alert alert-danger">{this.state.error}</div>: ""}

                       <form className="form-horizontal"  onSubmit={this.resetPasswordHandler}>
                           <div className="form-group">
                               <label className="control-label col-sm-2">Email:</label>
                               <div className="col-sm-10">
                                   <input type="email" name="email" className="form-control" placeholder="Enter email" 
                                    value={this.state.email} onChange={this.inputHandler}/>
                               </div>
                           </div>
                           <div className="modal-footer">
                               <button type="button" className="btn btn-secondary" data-dismiss="modal">Close</button>
                               <button type="submit" className="btn btn-primary">Send Email</button>
                           </div>
                       </form>
                   </div>
               </div>
           </div>
       </div>
        );
    }
}
export default ResetPassword;
