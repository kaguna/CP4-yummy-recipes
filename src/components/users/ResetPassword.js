/**
 * This is a modal component reset a password via email.
 */
import React, { Component } from 'react';
import Loader from 'react-loader-spinner';
import axiosInstance from '../../common/AxiosInstance';

class ResetPassword extends Component {
  constructor(props) {
    super(props);
    this.state = { errorMessage: '', successMessage: '', status: 'hidden' };
  }

    inputHandler = (event) => {
      const { name, value } = event.target;
      this.setState({ [name]: value });
    };

    resetPasswordHandler = (event) => {
      const email = { email: event.target.elements.emailAdress.value };
      event.preventDefault();
      this.setState({ status: '' });
      axiosInstance.post('/auth/send_reset_password_token', email)
        .then((response) => {
          this.setState({ successMessage: response.data.message,
            errorMessage: '',
            status: 'hidden' });
        })
        .catch((error) => {
          if (error.response) {
            this.setState({ errorMessage: error.response.data.message,
              successMessage: '',
              status: 'hidden' });
          } else if (error.request) {
            this.setState({ errorMessage: 'Cant connect to the server.Please check your connection and try again.',
              successMessage: '',
              status: 'hidden' });
          }
        });
      event.target.elements.emailAdress.value = ' ';
    };
    resetHandler = (event) => {
      this.setState({ successMessage: '', errorMessage: '' });
    }
    render() {
      const { successMessage, errorMessage, status } = this.state;
      return (
        <div
          className="modal fade"
          id="resetpassword"
          tabIndex="-1"
          role="dialog"
          aria-labelledby="exampleModalCenterTitle"
          aria-hidden="true"
        >
          <div className="modal-dialog" role="document">
            <div className="modal-content">
              <div className="modal-header">
                <h5 className="modal-title" id="exampleModalLongTitle">Email login credetials</h5>
                <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                  <span aria-hidden="true">&times;</span>
                </button>
              </div>
              <div className="modal-body">
                <div className={status}>
                  <Loader type="ThreeDots" color="#337ab7"height="100" width="100" />
                </div>
                {successMessage ?
                  <div className="alert alert-success">{successMessage}</div> : ''}
                {errorMessage ?
                  <div className="alert alert-danger">{errorMessage}</div> : ''}
                <form className="form-horizontal" onSubmit={this.resetPasswordHandler}>
                  <div className="form-group">
                    <label className="control-label col-sm-2">Email:</label>
                    <div className="col-sm-10">
                      <input
                        type="email"
                        name="emailAdress"
                        className="form-control"
                        placeholder="Enter email"
                        value={this.state.email}
                        onChange={this.inputHandler}
                      />
                    </div>
                  </div>
                  <div className="modal-footer">
                    <button type="button" className="btn btn-danger" onClick={this.resetHandler} data-dismiss="modal">Close</button>
                    <button type="submit" className="btn btn-primary">Send Email</button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      );
    }
}
export default ResetPassword;
