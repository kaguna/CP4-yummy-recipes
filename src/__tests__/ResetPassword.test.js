import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ResetPassword from '../components/users/ResetPassword';

describe('Reset password component', () => {
  const wrapper = shallow(<ResetPassword />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});