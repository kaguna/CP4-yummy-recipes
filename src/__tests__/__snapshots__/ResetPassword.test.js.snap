// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset password component renders properly 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalCenterTitle"
  className="modal fade"
  id="resetpassword"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLongTitle"
        >
          Email login credetials
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <div
          className="hidden"
        >
          <Loader
            color="#337ab7"
            height="100"
            type="ThreeDots"
            visible={true}
            width="100"
          />
        </div>
        <form
          className="form-horizontal"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              className="control-label col-sm-2"
            >
              Email:
            </label>
            <div
              className="col-sm-10"
            >
              <input
                className="form-control"
                name="emailAdress"
                onChange={[Function]}
                placeholder="Enter email"
                type="email"
              />
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-danger"
              data-dismiss="modal"
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
            <button
              className="btn btn-primary"
              type="submit"
            >
              Send Email
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
